apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts

```diff
@@ -122,6 +137,64 @@ export class ApplicationAccessService {
     return true
   }
 
+  private async hasValidCustomDelegation(
+    user: User,
+    template: ApplicationTemplate<
+      ApplicationContext,
+      ApplicationStateSchema<EventObject>,
+      EventObject
+    >,
+    scopeCheck?: boolean,
+  ): Promise<boolean> {
+    const userDelegations = user.delegationType
+    const requiredScopes = template.requiredScopes
+
+    // Must be a custom delegation
+    if (
+      !userDelegations ||
+      !userDelegations.includes(AuthDelegationType.Custom)
+    ) {
+      return false
+    }
+
+    // Custom delegations must have required scopes on their template
+    if (!requiredScopes) {
+      return false
+    }
+
+    // If we dont have the scopes available like in service portal we need to fetch
+    // them for the user logged in
+    if (scopeCheck) {
+      const delegations = await this.actorDelegationsApi
```

Comment:
We should consider moving this API call to the client library, creating a separate service for better reusability.

Reply:
I agree, would be happy to implement it once this has been merged to main as there is some pressure to get this in before the freeze.